miden-lib/build.rs

```diff
@@ -17,10 +69,42 @@ fn main() -> io::Result<()> {
     // or its builder changed:
     println!("cargo:rerun-if-changed=asm");
 
+    // Copies the Masm code to the build directory
+    let crate_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let build_dir = env::var("OUT_DIR").unwrap();
+    let src = Path::new(&crate_dir).join("asm");
+    let dst = Path::new(&build_dir).to_path_buf();
+    copy_directory(src, &dst);
+
+    // if this build has the testing flag set, modify the code and reduce the cost of proof-of-work
+    match env::var("CARGO_FEATURE_TESTING") {
+        Ok(ref s) if s == "1" => {
+            let constants = dst.join("asm/sat/internal/constants.masm");
+            let patched = dst.join("asm/sat/internal/constants.masm.patched");
+
+            // scope for file handlers
+            {
+                let read = File::open(&constants).unwrap();
+                let mut write = File::create(&patched).unwrap();
+                let modified = BufReader::new(read).lines().map(decrease_pow);
+
+                for line in modified {
+                    write.write_all(line.unwrap().as_bytes()).unwrap();
+                    write.write_all(&[b'\n']).unwrap();
+                }
+                write.flush().unwrap();
+            }
+
+            fs::remove_file(&constants).unwrap();
+            fs::rename(&patched, &constants).unwrap();
+        }
+        _ => (),
+    }
+
     let namespace =
         LibraryNamespace::try_from("miden".to_string()).expect("invalid base namespace");
     let version = Version::try_from(env!("CARGO_PKG_VERSION")).expect("invalid cargo version");
-    let stdlib = MaslLibrary::read_from_dir(ASM_DIR_PATH, namespace, false, version)?;
+    let stdlib = MaslLibrary::read_from_dir(dst.join("asm"), namespace, false, version)?;
 
     stdlib.write_to_dir(ASL_DIR_PATH)?;
```

Comment:
Should we also write the result into the `target` directory as we discussed on Discord?